crates/router/src/lib.rs

```diff
@@ -196,11 +199,52 @@ pub async fn start_server(conf: settings::Settings<SecuredSecret>) -> Applicatio
     );
     let state = Box::pin(AppState::new(conf, tx, api_client)).await;
     let request_body_limit = server.request_body_limit;
-    let server = actix_web::HttpServer::new(move || mk_app(state.clone(), request_body_limit))
-        .bind((server.host.as_str(), server.port))?
-        .workers(server.workers)
-        .shutdown_timeout(server.shutdown_timeout)
-        .run();
+
+    let server_builder =
+        actix_web::HttpServer::new(move || mk_app(state.clone(), request_body_limit))
+            .bind((server.host.as_str(), server.port))?
+            .workers(server.workers)
+            .shutdown_timeout(server.shutdown_timeout);
+
+    #[cfg(feature = "tls")]
+    let server = match server.tls {
```

Comment:
Can you please raise `ApplicationError` instead of the `unwrap()`s here? You can consider adding an enum variant along the lines of `TlsConfigurationError(String)`, and include suitable error messages in each case.

Reply:
Sorry, read the docstring and thought the function panics already, so can use `unwrap()`, turns out no unwraps there.
Do you want me to add a new variant or can I use 
```rust
#[error("Invalid configuration value provided: {0}")]
InvalidConfigurationValueError(String),
```
and just specify there

Reply:
I'm fine with either.